spring-integration-security/src/main/java/org/springframework/integration/security/channel/SecurityContextPropagationChannelInterceptor.java

```diff
@@ -87,8 +85,9 @@ public static void cleanup() {
 				SecurityContextHolder.setContext(originalContext);
 			}
 		}
-		catch (Throwable t) {
+		catch (Throwable t) {//NOSONAR - ok, we re-throw below
 			SecurityContextHolder.clearContext();
+			throw t;
```

Comment:
OK. Won't mind. Since we may have an issue here with only the `SecurityContextHolder`, the same must appear in the other place during the execution.
So, that `SONAR` comment must be fixed as well.